site/src/quacs-rs/src/lib.rs

```diff
@@ -26,254 +24,258 @@ pub fn init() {
     console_error_panic_hook::set_once();
 }
 
-#[wasm_bindgen(js_name = "generateSchedulesAndConflicts")]
-pub fn generate_schedules_and_conflicts() -> usize {
-    let selected_courses_map = SELECTED_COURSES.read().unwrap();
-    let mut selected_courses: Vec<&HashSet<u32>> = selected_courses_map
-        .values()
-        .filter(|set| !set.is_empty())
-        .collect();
+#[wasm_bindgen]
+impl Context {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    #[wasm_bindgen(js_name = "generateSchedulesAndConflicts")]
+    pub fn generate_schedules_and_conflicts(&mut self) -> usize {
+        let mut selected_courses: Vec<&HashSet<u32>> = self
+            .selected_courses
+            .values()
+            .filter(|set| !set.is_empty())
+            .collect();
+
+        let schedules_len = if selected_courses.is_empty() {
+            console_log!(
+                "Called generateSchedulesAndConflicts with no schedules, short circuiting!"
+            );
+
+            self.schedules = Vec::new();
+            self.curr_times = [0; BIT_VEC_LEN];
 
-    let schedules_len = if selected_courses.is_empty() {
-        console_log!("Called generateSchedulesAndConflicts with no schedules, short circuiting!");
+            0
+        } else {
+            selected_courses.sort_by_key(|set| set.len());
 
-        *SCHEDULES.write().unwrap() = Vec::new();
-        *CURR_TIMES.write().unwrap() = [0; BIT_VEC_LEN];
+            bm_start!("generate schedules");
+            self.curr_times = [u64::MAX; BIT_VEC_LEN];
+            let schedules =
+                Self::generate_schedules_driver(&mut selected_courses, &mut self.curr_times);
+            bm_end!("generate schedules");
 
-        0
-    } else {
-        selected_courses.sort_by_key(|set| set.len());
+            let schedules_len = schedules.len();
 
-        bm_start!("generate schedules");
-        #[allow(unused_assignments)]
-        let mut times = CURR_TIMES.write().unwrap();
-        *times = [u64::MAX; BIT_VEC_LEN];
-        let schedules = generate_schedules_driver(&mut selected_courses, &mut times);
-        bm_end!("generate schedules");
+            self.schedules = schedules;
 
-        let schedules_len = schedules.len();
+            schedules_len
```

Comment:
It might be cleaner to just return `self.schedules.len()` instead of storing the length beforehand.  Originally this was stored in a separate variable since it's annoying to read from the global variable, but with a member variable it's fine.

Reply:
Btw the diff for this one got messed up, this comment applies only to the new lines (GitHub thinks that the old file is offset for some reason)

Reply:
Fixed